### ISSUE TITLE
Add support for ignores for JSON and YAML files

### ISSUE THREAD

**Issue by cjcarpen** [jdkato/txtlint#383]
We'd like to be able to leverage our Vale styles to check for content outside of the docs, namely UI strings in JSON and YAML files. To clean up the results, particularly around spelling, we'd like to be able to configure in the `.vale.ini` file to ignore the keys in the files using regex.

```json
{
    "section_devtools": {
        "api": "Learn all about the Acme Test API.",
        "cli": "Run commands to manage your anvils by using the Acme Test CLI."
    },
    "section_recommended": [
        {
            "title": "Create an anvil",
            "description": "Create an anvil to drop on an unsuspecting passerby.",
            "href": "https://example.com/anvils"
        },
        {
            "title": "Create a sign",
            "description": "Create a sign to bamboozle your adversary.",
            "href": "https://example.com/signs"
        }
    ]
}
```

While Vale can currently run on JSON and YAML files, through our testing, it seems that token ignores are not currently supported for these file types. For example, even if I set an ignore that's simply for `devtools` exactly (no real regex), this is still flagged in the JSON but not if I put it in a Markdown file. 

After some troubleshooting, I searched around in the existing issues and found a comment from about a year ago that noted that token ignores only work for the supported Markup languages: https://github.com/errata-ai/vale/issues/289#issuecomment-742284453 So, if this is currently unsupported (and not just a goof on my part trying to get this configured), this issue is to request support for ignores in JSON and YAML files.

**Comment by jdkato** [comment]
Yes, `TokenIgnores` and `BlockIgnores` are only available in the supported markup formats.

My advice for data-serialization formats is to create a small wrapper (in any scripting language) that parses the file and passes only the desired keys to Vale. So, for example, you could lint `title` and `description` but ignore `href`.

I wrote about a related topic in [Vale & The OpenAPI Specification](https://jdkato.medium.com/vale-the-openapi-specification-8a7cfae135fb?sk=8fdfc2d058d997348a5cd30a31dfd834).

**Comment by cjcarpen** [comment]
Thank you so much for the response - the blog post is super helpful!
